enable_disable_persistence.md

```diff
@@ -0,0 +1,185 @@
+# Enable/Disable State Persistence Spec
+
+## Background
+
+Tilt currently has a system for changing the arguments to `tilt up` at
+runtime. We usually call it `tilt args`.
+
+https://docs.tilt.dev/tiltfile_config.html
+
+Most teams that use this system are using it for enabling and disabling sets of
+services, as demonstrated in this example:
+
+https://docs.tilt.dev/tiltfile_config.html#run-a-defined-set-of-services
+
+This helps save CPU when you only need to run a subset of your services.
+
+A number of teams have built wrappers around `tilt args`. This has helped us to
+learn a lot about the ergonomics that people want, and the pains of the current
+system. We have mocks and a pretty good idea of how we want the narrowly-focused
+single-click enable/disable interaction to work.
+
+For more details, see [Enable/Disable Resource API Spec](./enable_disable_resource.md).
+
+But we have different ideas on how we want the enable/disable interaction to work
+across multiple Tilt runs, and how the state should persist! [^1]
+
+## Primary Goal
+
+The goal of this doc is to give a product specification by example: describe
+specific sequences of user actions, and how we want Tilt to behave. (Note that
+it is not intended as a description of how Tilt behaves today.)
+
+## Non-Goals
+
+### Groups
+
+We expect that we will eventually want to control groups of services, but
+we don't believe that changes our analysis here about how single services behave.
+
+## Terminology
+
+### Enabled
+
+We say a server is "enabled" if it's visible in the Tilt UI and is in status
+waiting on trigger, pending on dependency, running, completed, or error.
+
+### Disabled
+
+We say a server is "disabled" if it's visible in the Tilt UI and is in status
+"disabled". We do not expect any new logs or status changes in the UI until the
+service is intentionally "enabled"
```

Comment:
my current understanding of the decision is that we want tilt to know about all services in your tiltfile (so you can list them, search them, etc). And that hiding them is a purely UI-level action. Though Surbhi / L can add more color on this.

Reply:
to get at both of these comments and the spec description, i think that the question of whether or not disabled services are visible to the user and to what extent they're visible (through the web app or through the cli) is a design and ux question that can be considered separate from how we define a "disabled" service. 

the reason i say that is because the design could change, but the state of the resource doesn't, ie, a disabled resource is one where the resource status is the equivalent of doing `tilt down <this resource>`. (i may not be using all the right terminology here, so lmk if i'm not making sense!) 

i'd also +1 for resolving the conflicting definitions of a disabled service, even if that's just internal for the moment and doesn't result in any code changes.

here's what i mean by removing the ui definition:
```suggestion
### Disabled

We say a server is "disabled" if it's described by the Tilt API and its status
"disabled". We do not expect any new logs or status changes until the
service is intentionally "enabled"
```

Reply:
ya, i like that!